kernel/tests/cdf.rs

```diff
@@ -0,0 +1,196 @@
+use std::{error, sync::Arc};
+
+use arrow::compute::filter_record_batch;
+use arrow_array::RecordBatch;
+use delta_kernel::engine::sync::SyncEngine;
+use itertools::Itertools;
+
+use delta_kernel::engine::arrow_data::ArrowEngineData;
+use delta_kernel::{DeltaResult, Table, Version};
+
+mod common;
+use common::{load_test_data, to_arrow};
+
+fn read_cdf_for_table(
+    test_name: impl AsRef<str>,
+    start_version: Version,
+    end_version: impl Into<Option<Version>>,
+) -> DeltaResult<Vec<RecordBatch>> {
+    let test_dir = load_test_data("tests/data", test_name.as_ref()).unwrap();
+    let test_path = test_dir.path().join(test_name.as_ref());
+    let table = Table::try_from_uri(test_path.to_str().expect("table path to string")).unwrap();
+    let engine = Arc::new(SyncEngine::new());
+    let table_changes = table.table_changes(engine.as_ref(), start_version, end_version)?;
+
+    // Project out the commit timestamp since file modification time may change anytime git clones
+    // or switches branches
+    let names = table_changes
+        .schema()
+        .fields()
+        .map(|field| field.name())
+        .filter(|name| *name != "_commit_timestamp")
+        .collect_vec();
+    let schema = table_changes.schema().project(&names)?;
+    let scan = table_changes
+        .into_scan_builder()
+        .with_schema(schema)
+        .build()?;
+    let batches: Vec<RecordBatch> = scan
+        .execute(engine)?
+        .map(|scan_result| -> DeltaResult<_> {
+            let scan_result = scan_result?;
+            let mask = scan_result.full_mask();
+            let data = scan_result.raw_data?;
+            let record_batch = to_arrow(data)?;
+            match mask {
+                Some(mask) => Ok(filter_record_batch(&record_batch, &mask.into())?),
+                None => Ok(record_batch),
+            }
+        })
+        .try_collect()?;
+    Ok(batches)
+}
+
+fn assert_batches_sorted_eq(expected_lines: &[impl ToString], batches: &[RecordBatch]) {
```

Comment:
is this duplicated? can we share code or just create follow up to clean this up soon?

Reply:
Ah good catch. Moved it to common and updated tests. We still passing 😎